### PR TITLE
Update confidence.rst

### PR DESCRIPTION
**Pull request by skoldobskiy**
Hello!
I found a bug in an example with calculating c.i.
The assessment of sigma's and related plotting of confidence intervals performed with sqrt(chi2/redchi), however, it should be sqrt((chi2-chi2.min)/redchi). The updated equation provides accurate and adequate contour plots for parameter uncertainties.

Also renaming dchi2 to chi2 can be considered.

#### Description
 
https://lmfit.github.io/lmfit-py/confidence.html#lmfit.conf_interval2d -- no ellipses are plotted here, however, they should. The proposed change corrects this.

 no
 no

###### Type of Changes

- [ x] Bug fix
- [ ] New feature
- [ ] Refactoring / maintenance
- [ ] Documentation / examples

###### Verification 
Have you

- [ ] included docstrings that follow PEP 257?

- [ ] referenced existing Issue and/or provided relevant link to mailing list?

- [ ] verified that existing tests pass locally?

- [ ] verified that the documentation builds locally?

- [ ] squashed/minimized your commits and written descriptive commit messages?

- [ ] added or updated existing tests to cover the changes?
- [ ] updated the documentation and/or added an entry to the release notes (doc/whatsnew.rst)?
- [ ] added an example?

### CODE REVIEW

**Comment by newville** [comment]
@skoldobskiy Ah, good catch.  We do that subtraction with `conf_interval2d(....., chi2_out=False)` but not with 
`conf_interval2d(....., chi2_out=True)`.   I guess it is easier to change the doc than the code, especially because `chi2.min()` is sometimes an interesting thing to know. 

So, I'm +1 on this change.   I'll leave this open for a couple of days in case there are any objections or other considerations.   But, I hope to merge this and #941 and then be ready to push a release soonish.

**Comment by skoldobskiy** [comment]
@newville Also I propose that the [https://lmfit.github.io/lmfit-py/confidence.html#lmfit.conf_interval2d] can be changed in a way to emphasize that conf_interval2d studies the ellipses of standard errors, not those obtained by `conf_interval(out, out, sigmas=sigma_levels)` (if I understood the procedures correctly).
Currently, this is not clear to the reader/user.

**Comment by skoldobskiy** [comment]
1. I found that [https://lmfit.github.io/lmfit-py/examples/documentation/confidence_chi2_maps.html#sphx-glr-examples-documentation-confidence-chi2-maps-py](https://lmfit.github.io/lmfit-py/examples/documentation/confidence_chi2_maps.html#sphx-glr-examples-documentation-confidence-chi2-maps-py) is corrected already. I updated my old version with this example to synchronize them.
2. I also have added a couple of sentences in the confidence interval discussion -- for clarification. However I would like if somebody can check that I am fully correct

**Comment by newville** [comment]
@skoldobskiy it looks like there are some RST formatting issues - like single quotes where you might want double backticks. Have you installed and run "pre-commit"?  That might help identify those issues.

File: doc/confidence.rst [hunk 1]
```diff
@@ -198,22 +203,30 @@ using :func:`conf_interval2d`:
...
-        # sigma matrix: sigma increases chi_square
-        # from  chi_square_best
-        # to    chi_square + sigma**2 * reduced_chi_square
-        # so:   sigma = sqrt(dchi2 / reduced_chi_square)
-        sigma_mat = np.sqrt(abs(dchi2_mat)/out.redchi)
-
-        # you could calculate the matrix of probabilities from sigma as:
-        # prob_mat  = np.erf(sigma_mat/np.sqrt(2))
+        if explicitly_calculate_sigma:
+            print("Generating chi-square map for ", pairs)
```
Comment (reneeotten) [inline doc/confidence.rst:214]: please use a formatting string

Comment (skoldobskiy) [reply doc/confidence.rst:214]: @newville @reneeotten Ok thanks, I'll check and fix it.

**Comment by newville** [comment]
@skoldobskiy Thanks -- this looks okay to me.  @reneeotten any objection from anyone to Squash+Merge?  If I don't hear otherwise, I will plan to do that on Monday.

**Comment by newville** [comment]
@skoldobskiy @reneeotten Thanks!

**Pull request merged**